lib/game.rb

```diff
@@ -0,0 +1,84 @@
+#no structure yet, just drafting some logic
+#will split up into dif methods in the future
+
+class Game
+  attr_reader :guess
+  attr_reader :code
+  attr_reader :guesses_left
+
+  def initialize()
+    @guesses_left = 10
+    @code = gen_secret_code
+    @guess = ""
+
+  end
+
+  def take_input()
+    print("What is your guess?\n\t>>>")
+    the_input = gets.split(" ")
+    #print(the_input + "\n")
+
+    if(check_input(the_input))
+      @guess = the_input
+      @guesses_left -= 1
+    else
+      puts("Sorry, that's not the correct input, please refer to the instruction for the input method")
+      take_input()
+    end
+  end
+
+  def gen_secret_code
+    randomness = Random.new
+    @code = Array.new(4, 0)
```

Comment:
There's a bit of a circular declaration going on. `@code = gen_secret_code` and in there @code = `Array.new(4, 0)`...

You don't need to reference @code/code at all in this method. You can just do 
`(1..4).map { |_| randomness.rand(1..6) }` as the entire function - the randomness declaration. The return of the function is being set to @code and the map will eval into the array you want